packages/react-network/src/server.ts

```diff
@@ -0,0 +1,74 @@
+import {Context} from 'koa';
+import {Header, StatusCode, CspDirective} from '@shopify/network';
+import {Manager} from './manager';
+
+export class ServerManager implements Manager {
+  private statusCodes: StatusCode[] = [];
+  private csp = new Map<CspDirective, string[] | boolean>();
+  private redirectUrl?: string;
+
+  redirectTo(url: string, status = StatusCode.Found) {
+    this.addStatusCode(status);
+    this.redirectUrl = url;
+  }
+
+  addStatusCode(statusCode: StatusCode) {
+    this.statusCodes.push(statusCode);
+  }
+
+  addCspDirective(directive: CspDirective, value: string | string[] | boolean) {
+    const normalizedValue = typeof value === 'string' ? [value] : value;
+    const newValue = Array.isArray(value)
+      ? [...(this.csp.get(directive) || []), ...normalizedValue]
+      : normalizedValue;
+
+    this.csp.set(directive, newValue);
+  }
+
+  extract() {
+    return {
+      status:
+        this.statusCodes.length > 0
+          ? this.statusCodes.reduce((large, code) => Math.max(large, code), 0)
+          : undefined,
+      csp: [...this.csp.entries()].reduce<{
+        [key: string]: string[] | string | boolean;
+      }>((csp, [directive, value]) => ({...csp, [directive]: value}), {}),
+      redirectUrl: this.redirectUrl,
+    };
+  }
+}
+
+export function applyToContext<T extends Context>(
```

Comment:
I'd like to see this broken out into a different file so that a consumer doesn't need to pull in Koa if they're not using Koa/this function.

Reply:
This is already only available via the server entry point, and Koa is the only server library we really support. This also only bring in the typings for Koa, it does not bring in the library itself.